eg/Classes/PluginBase.py

```diff
@@ -148,7 +148,15 @@ def AddActionsFromList(self, theList, defaultAction=None):
         self.info.actionGroup.AddActionsFromList(theList, defaultAction)
 
     def AddEvents(self, *eventList):
-        self.info.eventList = eventList
+        if self.info.eventList is None:
+            self.info.eventList = ()
+
+        for event in eventList:
+            if isinstance(event, basestring):
+                event = (event, '')
+            elif isinstance(event, list):
+                event = tuple(event)
+            self.info.eventList += (event,)
```

Comment:
so if you come across an event and want to add it... all you would have to do is pass the eventstring to AddEventList..

now i should have it check for possible duplicates. that would also be beneficial. because if you use this method from multiple spots you won't have to add repeat code. just pass the event and let the method figure out if it should add it or not...

a dictionary might be a better container instead of a tuple for this... but i do not know how much code you would have to change with the AddEvents dialog to account for this. 

maybe the EventInfo class is what should really be used to house the data..  and can be iterated over to spit out keys and values of a dict that holds all of the events. set it up to do this out of the gate.. then you won't have to add them to add each event to an EventInfo class each time the AddEvent dialog loads.